### DISCUSSION TITLE
Using the Stencil Shapes to display data.

### DISCUSSION THREAD

**Discussion by Fractal393** [clientIO/joint#2504] [Q&A]
### Introduction

I am building an angular application.
I have created a stencil like a service similar to the one in the kitchen-sink demo and i am calling the shapes present in the stencil into my main component where i get the data from the api and use the stencil to render shapes. I am able to render shapes but why i try to embed a few shapes into a shape. It does not happen. What is the best way to use the shapes from the stencil and apply in on the data to render it?

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS, JointJS+

**Comment by kumilingus** [comment]
I am not sure I follow.
You mentioned `embedding`. Is your question related to https://github.com/clientIO/joint/discussions/2126?
